### PR TITLE
[Rest Api Compatibility] Move won't fix tests from block list

### PR DESCRIPTION
**Pull request by pgomulka**
block list at the moment contains a lot of tests that we already identified that won't be fixed.
This commit moves them out of the main block list so that it is easier to count progress.

### CODE REVIEW

**Comment by pgomulka** [comment]
my groovy is not best :D let me know how to improve this

File: rest-api-spec/build.gradle [hunk 1]
```diff
@@ -28,100 +28,102 @@ testClusters.all {
...
+          'cat.snapshots/10_basic/Test cat snapshots output',
+          'cat.tasks/10_basic/Test cat tasks output',
+          'cat.tasks/10_basic/Test cat tasks output with X-Opaque-Id',
+          'cat.templates/10_basic/Column headers',
+          'cat.templates/10_basic/Filtered templates',
+          'cat.templates/10_basic/Mixture of legacy and composable templates',
+          'cat.templates/10_basic/Normal templates',
+          'cat.templates/10_basic/Select columns',
+          'cat.thread_pool/10_basic/Test cat thread_pool output',
+          // type information about the type is removed and not passed down. The logic to check for this is also removed.
```
Comment (mark-vieira) [inline rest-api-spec/build.gradle:90]: Should we cleanup/remove some of these comments then given we don't plan on addressing any of these?

Comment (pgomulka) [reply rest-api-spec/build.gradle:90]: I think those are beneficial. It is good to know the reason why the test is disabled without actually reading the test. The reasoning why it is disabled is clear now, but after a month or two (or earlier :D ) we might forget the details

**Comment by pgomulka** [comment]
@elasticmachine run Check labels

**Review by mark-vieira — APPROVED** [review]
:+1:

File: rest-api-spec/build.gradle [hunk 2]
```diff
@@ -28,100 +28,102 @@ testClusters.all {
 
 tasks.named("test").configure { enabled = false }
 tasks.named("jarHell").configure { enabled = false }
-
+def willNotFixTests = {
+  return ['cat.aliases/10_basic/Alias against closed index',
```
Comment (jakelandis) [inline rest-api-spec/build.gradle:32]: I think we can use wildcards in black lists. I would suggest to wildcard as much of these cat tests as possible to help with brevity and future cases. Also a brief mention that compatibility is not supported for cat APIs would help

File: rest-api-spec/build.gradle [hunk 3]
```diff
@@ -28,100 +28,102 @@ testClusters.all {
 
 tasks.named("test").configure { enabled = false }
 tasks.named("jarHell").configure { enabled = false }
-
+def willNotFixTests = {
```
Comment (jakelandis) [inline rest-api-spec/build.gradle:31]: nit: s/willNotFixTests/v7compatiblityNotSupportedTests

**Review by jakelandis — APPROVED** [review]
LGTM

**Pull request merged**